homeassistant/components/openweathermap/repairs.py

```diff
@@ -0,0 +1,31 @@
+"""Issues for OpenWeatherMap."""
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import issue_registry as ir
+
+from .const import DOMAIN
+
+
+def _get_issue_id(entry_id: str) -> str:
+    return f"deprecated_v25_{entry_id}"
+
+
+@callback
+def async_create_issue(hass: HomeAssistant, entry_id: str) -> None:
+    """Create issue for V2.5 deprecation."""
+    ir.async_create_issue(
+        hass=hass,
+        domain=DOMAIN,
+        issue_id=_get_issue_id(entry_id),
+        is_fixable=True,
```

Comment:
It looks like the repair flow to switch to 3.0 still needs to be implemented (or maybe it didn't get pushed?)

https://github.com/home-assistant/core/blob/dev/homeassistant/components/kitchen_sink/repairs.py

Reply:
Flow to automatically migrate to 3.0?
I think, in this case, manual migration is better as it encourage user to perform additional steps (described in documentation) before the migration, which are necessary.
But if you think repair flow is required, I can do that.

Reply:
I think the repair flow is a better experience as users frequently have trouble reconfiguring the config entry on their own, and since the repair flow pops up and can walk them through the process they should be able to get it right the first time.

Reply:
do i need to check is api key valid (like in config_flow) in repair flow?

Reply:
Yes, I think so. Probably break that out into a utility function so it can be used in both places

Reply:
just added